src/base/PolymeshTransactionBase.ts

```diff
@@ -251,79 +260,93 @@ export abstract class PolymeshTransactionBase<
    *   throwing any pertinent errors
    */
   private async internalRun(): Promise<ISubmittableResult> {
-    const { signingAddress, signer, context } = this;
+    const { signingAddress, signer, mortality, context } = this;
+
     const nonce = context.getNonce().toNumber();
+    // era represents how many blocks the transaction should be valid for
+    const era = mortality.immortal ? 0 : mortality.blocksToLive?.toNumber();
+
     this.updateStatus(TransactionStatus.Unapproved);
 
     return new Promise((resolve, reject) => {
       const txWithArgs = this.composeTx();
       let settingBlockData = Promise.resolve();
-      const gettingUnsub = txWithArgs.signAndSend(signingAddress, { nonce, signer }, receipt => {
-        const { status } = receipt;
-        let isLastCallback = false;
-        let unsubscribing = Promise.resolve();
-        let extrinsicFailedEvent;
 
-        // isCompleted === isFinalized || isInBlock || isError
-        if (receipt.isCompleted) {
-          if (receipt.isInBlock) {
-            const blockHash = status.asInBlock;
+      const gettingUnsub = txWithArgs.signAndSend(
+        signingAddress,
+        { nonce, signer, era },
```

Comment:
It seems like adding `era` here causes the formatting to cascade whitespace changes.

The only change here should be adding `era` to the options object here